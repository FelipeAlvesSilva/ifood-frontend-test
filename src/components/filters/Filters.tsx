import React, { useEffect, useState, Fragment } from 'react';
import { Collapse } from 'antd';
import { BsFilter } from 'react-icons/bs';
import { mockyAPI } from '../../services/mockyApi';
import { FiltersType, Filter } from '../../utils/filtersLabels';

type FilterPanelProps = {
  filterLabel: Filter
};

const FilterPanel = (props: FilterPanelProps): JSX.Element => {
  switch (props.filterLabel.id.toLocaleLowerCase()) {
    case ('locale'):
      return (
        <Fragment>locale</Fragment>
      );
    case ('country'):
      return (
        <Fragment>country</Fragment>
      );
    case ('timestamp'):
      return (
        <Fragment>timestamp</Fragment>
      );
    case ('limit'):
      return (
        <Fragment>limit</Fragment>
      );
    case ('offset'):
      return (
        <Fragment>offset</Fragment>
      );
    default:
      return (
        <Fragment></Fragment>
      ); // No filters available
  }
};

// UX - Clean filters option
const Filters = () => {
  const { Panel } = Collapse;
  const [filters, setFilters] = useState<FiltersType>({ labels: [] });

  useEffect(() => {
    mockyAPI.get('/')
      .then(res => {
        setFilters(filters => res.data.filters);
      });
  }, []);

  const extraIcon = (): JSX.Element => <BsFilter onClick={(e) => e.stopPropagation()} />;

  return (
    <Collapse bordered={false} expandIconPosition='right' className='m-3'>
      <Panel header='Filters' key='1' extra={extraIcon()}>
        {/* {filters.labels?.length > 0
          ? filters.labels.map(filter => <FilterPanel key={filter.id} filterLabel={filter} />) 
          : null} */}
      </Panel>
    </Collapse>
  )
};

export default Filters;