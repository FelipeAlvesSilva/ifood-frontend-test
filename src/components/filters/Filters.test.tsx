import React from 'react';
import { render } from '@testing-library/react';
import Filters from '../../components/filters/Filters';

describe('Filters component tests', () => {
  test('Should not regress', () => {
    const compRender = render(<Filters />);
    expect(compRender).toMatchSnapshot();
  });
});