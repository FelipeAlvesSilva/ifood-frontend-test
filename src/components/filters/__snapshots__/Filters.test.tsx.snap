// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters component tests Should not regress 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="ant-collapse ant-collapse-borderless ant-collapse-icon-position-right m-3"
      >
        <div
          class="ant-collapse-item"
        >
          <div
            aria-expanded="false"
            class="ant-collapse-header"
            role="button"
            tabindex="0"
          >
            <span
              aria-label="right"
              class="anticon anticon-right ant-collapse-arrow"
              role="img"
            >
              <svg
                aria-hidden="true"
                class=""
                data-icon="right"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                />
              </svg>
            </span>
            Filters
            <div
              class="ant-collapse-extra"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M6 10.5a.5.5 0 01.5-.5h3a.5.5 0 010 1h-3a.5.5 0 01-.5-.5zm-2-3a.5.5 0 01.5-.5h7a.5.5 0 010 1h-7a.5.5 0 01-.5-.5zm-2-3a.5.5 0 01.5-.5h11a.5.5 0 010 1h-11a.5.5 0 01-.5-.5z"
                  fill-rule="evenodd"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="ant-collapse ant-collapse-borderless ant-collapse-icon-position-right m-3"
    >
      <div
        class="ant-collapse-item"
      >
        <div
          aria-expanded="false"
          class="ant-collapse-header"
          role="button"
          tabindex="0"
        >
          <span
            aria-label="right"
            class="anticon anticon-right ant-collapse-arrow"
            role="img"
          >
            <svg
              aria-hidden="true"
              class=""
              data-icon="right"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
              />
            </svg>
          </span>
          Filters
          <div
            class="ant-collapse-extra"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M6 10.5a.5.5 0 01.5-.5h3a.5.5 0 010 1h-3a.5.5 0 01-.5-.5zm-2-3a.5.5 0 01.5-.5h7a.5.5 0 010 1h-7a.5.5 0 01-.5-.5zm-2-3a.5.5 0 01.5-.5h11a.5.5 0 010 1h-11a.5.5 0 01-.5-.5z"
                fill-rule="evenodd"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
